android-components/components/support/ktx/build.gradle

```diff
@@ -41,6 +41,7 @@ dependencies {
     implementation ComponentsDependencies.androidx_core_ktx
     implementation ComponentsDependencies.kotlin_stdlib
     implementation ComponentsDependencies.kotlin_coroutines
+    implementation ComponentsDependencies.google_material
```

Comment:
Drive-by 🚗 : A bit unfortunate to have a base support library need to depend on a UI dependency. It looks like we only need it for `Snackbar.LENGTH_LONG`?

Reply:
Agreed, I did think about changing SnackbarDelegate but as it already existed avoided making the changes. My thoughts were: 

- document the existing duration param to describe the accepted int vals e.g. -1 for short (open to user error)
- have a duration enum in snackbar delegate (felt like duplication of Snackbar)
- change duration to a boolean e.g true for long, false for short (doesn't seem particularly clean)
- have showShort and showLong funcs (my preference)

Something like this:


```
    override fun showLong(
        snackBarParentView: View,
        text: Int,
        duration: Int,
        action: Int,
        listener: ((v: View) -> Unit)?,
    ) {
        show(snackBarParentView, text, Snackbar.LENGTH_LONG, action, listener)
    }

    override fun showShort(
        snackBarParentView: View,
        text: Int,
        duration: Int,
        action: Int,
        listener: ((v: View) -> Unit)?,
    ) {
       show(snackBarParentView, text, Snackbar.LENGTH_SHORT, action, listener)
    }

    fun show(
        snackBarParentView: View,
        text: Int,
        duration: Int,
        action: Int,
        listener: ((v: View) -> Unit)?,
    ) {
        val snackbar = Snackbar.make(
            snackBarParentView,
            text,
            duration,
        )

        if (action != 0 && listener != null) {
            snackbar.setAction(action, listener)
        }

        snackbar.show()
    }
```

Interested to get your thoughts.

Reply:
The last option makes sense to me (assuming the `duration` param on the override methods are a typo). Where would those two overrides live though? In `feature-contextmenu` I guess (which is fine)?

Reply:
@jonalmeida yep typo! My thinking was to move it all within the SnackbarDelegate. I've moved this to another issue to unblock this https://bugzilla.mozilla.org/show_bug.cgi?id=1820467